MANIFEST.in

```diff
@@ -1,50 +1,100 @@
 # Reference: https://setuptools.pypa.io/en/latest/userguide/miscellaneous.html
 
-# Include source files in SDist
-include CMakeLists.txt
-include *.bzl *.bazel .bazel* BUILD *.BUILD BUILD.* WORKSPACE
-include BUCK BUCK.*
-include requirements*.txt
-include version.txt
-include [Mm]akefile *.[Mm]akefile [Mm]akefile.*
-include [Dd]ockerfile *.[Dd]ockerfile [Dd]ockerfile.* .dockerignore
+# Include individual top-level files
+include MANIFEST.in
+include CITATION.cff
+include CODEOWNERS
+include Dockerfile
+include LICENSE
+include Makefile
+include NOTICE
+include .clang-format .clang-tidy
+include .cmakelintrc
+include .coveragerc
+include .dockerignore
+include .flake8
+include .gdbinit
+include .lintrunner.toml
+include .lldbinit
+include docker.Makefile
+include ubsan.supp
+
+# Include bazel and BUCK related files
+include BUILD.bazel BUCK.oss
+include WORKSPACE
+include *.bzl
+include .bazelignore .bazelrc .bazelversion
+
+# Include general configuration files
+include *.ini
+# Include important top-level information
+include *.md
+# Include technical text files at the moment, comprises
+# version.txt, CMakeLists.txt, requirements.txt
+include *.txt
+
+# Include ctags configuration
+include .ctags.d/*.ctags
+
+# Include subfolders completely
+graft .devcontainer
+graft .vscode
 graft android
+# The following folder (assets) is empty except for a .gitignore file, which
+# will not be included in the sdist, hence we include the directory explicitly.
+include android/test_app/app/src/main/assets
 graft aten
 graft binaries
 graft c10
 graft caffe2
 graft cmake
-graft functorch
-graft third_party
-graft tools
 graft torch
+graft tools
+graft test
+graft docs
+graft ios
+graft third_party
+graft test
+graft benchmarks
+graft scripts
+graft mypy_plugins
+graft modules
+graft functorch
 graft torchgen
 # FIXME: torch-xla build during codegen will fail if include this file in wheel
 exclude torchgen/BUILD.bazel
 
-# Misc files and directories in SDist
-include *.md
-include CITATION.cff
-include LICENSE NOTICE
-include mypy*.ini
-graft benchmarks
-graft docs
-graft mypy_plugins
-graft scripts
+# The following exclusions omit parts from third-party dependencies that
+# contain invalid symlinks[1] and that are not needed for pytorch, such as
+# bindings for unused languages
+prune third_party/ittapi/rust
+prune third_party/flatbuffers/java
+prune third_party/flatbuffers/kotlin
+prune third_party/nccl/pkg/debian
+prune third_party/opentelemetry-cpp/third_party/prometheus-cpp/cmake/project-import-*
```

Comment:
These git submodules might not be initialized when creating the SDist.

Reply:
I would consider it an error if they were not initialized at the point of creation of the sdist, much like an unfinished checkout of the pytorch repo itself.

I am convinced that they should be included as this (and only this?) enables the reproducable full build.